packages/react-events/src/dom/Keyboard.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {
+  ReactDOMResponderEvent,
+  ReactDOMResponderContext,
+} from 'shared/ReactDOMTypes';
+
+import React from 'react';
+import {DiscreteEvent} from 'shared/ReactTypes';
+
+type KeyboardEventType = 'keydown' | 'keyup';
+
+type KeyboardListenerProps = {|
+  onKeyDown: (e: KeyboardEvent) => void,
+  onKeyUp: (e: KeyboardEvent) => void,
+|};
+
+type KeyboardResponderProps = {
+  disabled: boolean,
+};
+
+type KeyboardEvent = {|
+  altKey: boolean,
+  ctrlKey: boolean,
+  isComposing: boolean,
+  key: string,
+  location: number,
+  metaKey: boolean,
+  repeat: boolean,
+  shiftKey: boolean,
+  target: Element | Document,
+  type: KeyboardEventType,
+  timeStamp: number,
+|};
+
+const targetEventTypes = ['keydown', 'keyup'];
+
+/**
+ * Normalization of deprecated HTML5 `key` values
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent#Key_names
+ */
+const normalizeKey = {
```

Comment:
Which browsers still expose the deprecated names?

Reply:
I took this code from:

https://github.com/facebook/react/blob/master/packages/react-dom/src/events/getEventKey.js

I remember when working on React Fire that not supporting this broke many internal web driver tests, which use an older version of FF.

Reply:
Ok we should probably revisit this because to cut down on legacy code